node/src/components/storage/tests.rs

```diff
@@ -559,8 +574,18 @@ fn different_block_at_height_is_fatal() {
     let mut storage = storage_fixture(&harness);
 
     // Create two different blocks at the same height.
-    let block_44_a = random_block_at_height(&mut harness.rng, 44);
-    let block_44_b = random_block_at_height(&mut harness.rng, 44);
+    let block_44_a = Box::new(Block::random_with_specifics(
+        &mut harness.rng,
+        EraId::new(1),
+        44,
+        false,
+    ));
+    let block_44_b = Box::new(Block::random_with_specifics(
+        &mut harness.rng,
+        EraId::new(1),
+        44,
+        false,
+    ));
```

Comment:
Should we add a similar `should_panic` test for two switch blocks in the same era (but possibly different height)?

Reply:
In a separate PR?

Reply:
(I'm not sure if it's needed, but I created [NDRS-1198](https://casperlabs.atlassian.net/browse/NDRS-1198) anyway.)